crates/red_knot_python_semantic/resources/mdtest/comparison/instances/rich_comparison.md

```diff
@@ -120,24 +141,24 @@ class B:
 # Because `object.__eq__` and `object.__ne__` accept `object` in typeshed,
 # this can only happen with an invalid override of these methods,
 # but we still support it.
-reveal_type(B() == A())  # revealed: int
-reveal_type(B() != A())  # revealed: bytearray
+reveal_type(B() == A())  # revealed: EqReturnType
+reveal_type(B() != A())  # revealed: NeReturnType
 
-reveal_type(B() < A())  # revealed: list
-reveal_type(B() <= A())  # revealed: set
+reveal_type(B() < A())  # revealed: GtReturnType
+reveal_type(B() <= A())  # revealed: GeReturnType
 
-reveal_type(B() > A())  # revealed: str
-reveal_type(B() >= A())  # revealed: bytes
+reveal_type(B() > A())  # revealed: LtReturnType
+reveal_type(B() >= A())  # revealed: LeReturnType
```

Comment:
@AlexWaygood I found a few more instances here. Would you mind having a look at this particular test. I'm not quite sure what to do with the `other: str` annotations in `class B` above. Should those be changed to `LtReturnType`, or is `str` just *some unrelated type* here?

Reply:
> Should those be changed to `LtReturnType`, or is `str` just _some unrelated type_ here?

The latter -- `str` here is just "something that is not `object`, `A` or `B`". I think the best thing would be to create a new `class Unrelated: ...` definition and use that in the annotations for the `other` parameter in `B.__eq__` and `B.__ne__`.